tests/test_client.py

```diff
@@ -357,17 +347,16 @@ def test_save_to_well_known_file_service_account(self):
         self.assertEqual('ABCDEF', d['private_key_id'])
         os.remove(temp_credential_file)
 
-    def test_save_well_known_file_with_non_existent_config_dir(self):
+    @mock.patch('os.path.isdir', return_value=False)
+    def test_save_well_known_file_with_non_existent_config_dir(self,
```

Comment:
`@mock.patch` (which is now in the standard library in Python 3) adds sugar for more arguments. Also, I've used it in PRs that you've reviewed before.